src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Anonymizer.Core.Utility;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export;
+using Microsoft.Health.Fhir.Core.Models;
+using NSubstitute;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Core.UnitTests.Features.Operations.Export
+{
+    public class ExportAnonymizerTests
+    {
+        [Fact]
+        public async Task GivenRedactAnonymizationConfig_WhenAnonymizeResource_ThenPropertiesShouldBeRedacted()
+        {
+            string configurationContent =
+            @"
+            {
+	            ""fhirPathRules"": [
+		            {
+			            ""path"": ""Patient.name"",
+			            ""method"": ""redact""
+		            }
+	            ]
+            }";
+
+            Patient patient = new Patient();
+            patient.Name.Add(HumanName.ForFamily("Test"));
+            IAnonymizer anonymizer = await CreateAnonymizerFromConfigContent(configurationContent);
+
+            ResourceElement resourceElement = anonymizer.Anonymize(new ResourceElement(patient.ToTypedElement()));
+            Patient anonymizedResource = resourceElement.Instance.ToPoco<Patient>();
+            Assert.Empty(anonymizedResource.Name);
+        }
+
+        [Fact]
+        public async Task GivenGeneralizeAnonymizationConfig_WhenAnonymizeResource_ThenPropertiesShouldBeGeneralized()
+        {
+            string configurationContent =
+            @"
+            {
+              ""fhirPathRules"": [
+                {
+                  ""path"": ""Patient.birthDate"",
+                  ""method"": ""generalize"",
+                  ""cases"": {
+                    ""$this <= @2020-01-01 and $this >= @1990-01-01"": ""@2000-01-01""
+                  }
+                }
+              ]
+            }";
+
+            Patient patient = new Patient();
+            patient.BirthDate = "2001-01-01";
```

Comment:
nit: would it be useful to have a Patient resource whose birthdate false outside the range and validate that their birthdate was not changed?